scalameta/semanticdb-scalac/src/test/scala/scala/meta/tests/SemanticSuite.scala

```diff
@@ -1007,12 +1016,96 @@ class SemanticSuite extends DatabaseSuite(SemanticdbMode.Slim) {
       |  [47..50): Int => _root_.scala.Int#
-      |[178..178): *(scala.collection.immutable.IndexedSeq.canBuildFrom[Tuple2[Int, Int]])
+      |  [54..55): * => _star_.
+      |  [108..114): Tuple2 => _root_.scala.Tuple2#
+      |  [115..118): Int => _root_.scala.Int#
+      |  [120..123): Int => _root_.scala.Int#
+      |  [95..107): canBuildFrom => _root_.scala.collection.immutable.IndexedSeq.canBuildFrom()Lscala/collection/generic/CanBuildFrom;.
+    """.trim.stripMargin
+  )
+
+  synthetics(
+    """
+      |object ForCompFuture {
+      |  import scala.concurrent.ExecutionContext.Implicits.global
+      |  val f1 = scala.concurrent.Future {1}
+      |  val f2 = scala.concurrent.Future {2}
+      |
+      |  for (v1 <- f1; v2 <- f2 if v1 > v2) yield v1+v2
+      |}
+    """.trim.stripMargin,
+    """
+    |[117..117): *.apply[Int]
+    |  [0..1): * => _star_.
+    |  [2..7): apply => _root_.scala.concurrent.Future.apply(Lscala/Function0;Lscala/concurrent/ExecutionContext;)Lscala/concurrent/Future;.
+    |  [8..11): Int => _root_.scala.Int#
+    |[121..121): *(scala.concurrent.ExecutionContext.Implicits.global)
+    |  [0..1): * => _star_.
+    |  [46..52): global => _root_.scala.concurrent.ExecutionContext.Implicits.global.
+    |[156..156): *.apply[Int]
+    |  [0..1): * => _star_.
+    |  [2..7): apply => _root_.scala.concurrent.Future.apply(Lscala/Function0;Lscala/concurrent/ExecutionContext;)Lscala/concurrent/Future;.
+    |  [8..11): Int => _root_.scala.Int#
+    |[160..160): *(scala.concurrent.ExecutionContext.Implicits.global)
+    |  [0..1): * => _star_.
+    |  [46..52): global => _root_.scala.concurrent.ExecutionContext.Implicits.global.
+    |[177..177): *.flatMap[Int](*)(scala.concurrent.ExecutionContext.Implicits.global)
+    |  [0..1): * => _star_.
+    |  [2..9): flatMap => _root_.scala.concurrent.Future#flatMap(Lscala/Function1;Lscala/concurrent/ExecutionContext;)Lscala/concurrent/Future;.
+    |  [10..13): Int => _root_.scala.Int#
+    |  [15..16): * => _star_.
+    |  [62..68): global => _root_.scala.concurrent.ExecutionContext.Implicits.global.
+    |[187..187): *.withFilter(*)(scala.concurrent.ExecutionContext.Implicits.global)
+    |  [0..1): * => _star_.
+    |  [2..12): withFilter => _root_.scala.concurrent.Future#withFilter(Lscala/Function1;Lscala/concurrent/ExecutionContext;)Lscala/concurrent/Future;.
+    |  [13..14): * => _star_.
+    |  [60..66): global => _root_.scala.concurrent.ExecutionContext.Implicits.global.
+    |[198..198): *.map[Int](*)(scala.concurrent.ExecutionContext.Implicits.global)
+    |  [0..1): * => _star_.
+    |  [2..5): map => _root_.scala.concurrent.Future#map(Lscala/Function1;Lscala/concurrent/ExecutionContext;)Lscala/concurrent/Future;.
+    |  [6..9): Int => _root_.scala.Int#
+    |  [11..12): * => _star_.
+    |  [58..64): global => _root_.scala.concurrent.ExecutionContext.Implicits.global.
+   """.trim.stripMargin
+  )
+
+  synthetics(
+    """
+      |object ForeachFuture {
```

Comment:
We should try to keep the tests as narrow as possible, there is a lot of duplication from above. A single test like this should be enough to cover all cases

```scala
import scala.concurrent.ExecutionContext.Implicits.global
for (a <- scala.concurrent.Future.successful(1)) println(a)
for {
  a <- scala.concurrent.Future.successful(1)
  b <- scala.concurrent.Future.successful(2)
  if a < b
} yield a
```

Reply:
Shouldn't we do:

```scala
for {
  a <- scala.concurrent.Future.successful(1)
  b <- scala.concurrent.Future.successful(2)
} println(a)
```
to make sure we see two distinct injection of implicits for foreach?

Reply:
The first case covers foreach and the second case covers flatmap/map/withFilter, I don't think we can cover map and foreach in the same for comprehension.

Reply:
Aah, I see what you mean. Sorry! Yes we can put two generators to introduce a conflict in positions.